lib/auth0/api/v2/organizations.rb

```diff
@@ -0,0 +1,332 @@
+module Auth0
+  module Api
+    module V2
+      # Methods to use the organizations endpoints
+      module Organizations
+        include Auth0::Mixins::Validation
+
+        attr_reader :organizations_path
+
+        # Get all organizations.
+        # @see https://auth0.com/docs/api/management/v2/#!/Organizations/get_organizations
+        # @param options [hash] The Hash options used to define the paging of rersults
+        #   * :per_page [integer] The amount of entries per page. Default: 50. Max value: 100.
+        #   * :page [integer] The page number. Zero based.
+        #   * :include_totals [boolean] True to include query summary in the result, false or nil otherwise.
+        # @return [json] All Organizations
+        def organizations(options = {})
+          request_params = {
+            per_page:       options.fetch(:per_page, nil),
+            page:           options.fetch(:page, nil),
+            include_totals: options.fetch(:include_totals, nil)
+          }
+          get(organizations_path, request_params)
+        end
+        alias get_organizations organizations
+
+        # Create a new organization.
+        # @see ttps://auth0.com/docs/api/management/v2/#!/Organizations/post_organizations
+        # @param options [hash] See https://auth0.com/docs/api/management/v2/#!/Organizations/post_organizations for available options
+        # @return [json] Returns the created organization.
+        def create_organization(options = {})
+          post(organizations_path, options)
+        end
+
+        # Get an organization by id. A token with read:organizations scope is required
+        # @see https://auth0.com/docs/api/management/v2/#!/Organizations/get_organizations_by_id
+        # @param organization_id [string] The organization_id of the user to retrieve.
+        #
+        # @return [json] Returns the organization with the given organization_id if it exists.
+        def organization(organization_id)
+          raise Auth0::MissingOrganizationId, 'Must supply a valid organization_id' if organization_id.to_s.empty?
+          path = "#{organizations_path}/#{organization_id}"
+          get(path)
+        end
+
+        # Get an organization by name. A token with read:organizations scope is required.
+        # @see https://auth0.com/docs/api/management/v2/#!/Organizations/get_name_by_name
+        # @param organization_id [string] The Organization ID
+        #
+        # @return [json] Returns the organization with the given organization_id if it exists.
+        def organization_by_name(organization_name)
+          raise Auth0::MissingOrganizationId, 'Must supply a valid organization_name' if organization_name.to_s.empty?
```

Comment:
Is using `MissingOrganizationId` here appropriate for this method, which takes a name? Do we have one for `MissingOrganizationName`?